src/components/application_manager/src/application_manager_impl.cc

```diff
@@ -3824,6 +3823,16 @@ void ApplicationManagerImpl::OnUpdateHMIAppType(
   }
 }
 
+void ApplicationManagerImpl::OnPTUFinished(const bool ptu_result) {
```

Comment:
@LuxoftAKutsan 
There is PolicyHandlerObserver class which is already inherited by AM. This method is just overriden here to trigger resource releasing only in case PTU happened but not declined by SDL.